src/System.Threading/tests/MutexTests.cs

```diff
@@ -154,5 +156,51 @@ public void AbandonExisting()
                 Assert.Equal(m, ame.Mutex);
             }
         }
+
+        [PlatformSpecific(PlatformID.Windows)] // names aren't supported on Unix
+        [Fact]
+        public void CrossProcess_NamedMutex_ProtectedFileAccessAtomic()
+        {
+            string mutexName = Guid.NewGuid().ToString("N");
```

Comment:
It might be good to also test names prefixed with "Global\" and "Local\", perhaps as a `[Theory]`. https://msdn.microsoft.com/en-us/library/windows/desktop/aa382954(v=vs.85).aspx